### PR TITLE
Improve workspace load test automation

### LINKED ISSUE #9901: Workspaces started with loadgen can be gracefully terminated via gpctl stop

### PR DESCRIPTION
**Pull request by Furisto**
## Description
For a description of the changes and a demo, please see the Loom video. In short:
- Check success rate of started workspaces
- Introduce score to influence the likelihood of a repository used as context for the workspace
- Clean up workspaces after test has been performed
https://www.loom.com/share/6487e3403c0746cc97bb3f766e15fab6

## Related Issue(s)
Fixes #9901, #9722, #9724

## How to test
- Spin up a preview vm (or an ephemeral cluster)
- Start loadgen tool with loadgen benchmark prod-benchmark.yaml --tls [pathToTlsConfig] --host [addressOfWsManager] .Tls config can be retrieved with gpctl clusters get-tls-config (ensure you have the correct kubeconfig). You can port-forward ws-manager with e.g. kubectl port-forward [ws-manager-pod] 12001:8080

## Release Notes

```release-note
None
```

### CODE REVIEW

File: dev/loadgen/cmd/benchmark.go [hunk 1]
```diff
@@ -27,6 +29,8 @@ import (
 	"github.com/gitpod-io/gitpod/ws-manager/api"
 )
 
+const benchmarkAnnotation = "benchmark"
```
Comment (sagor999) [inline dev/loadgen/cmd/benchmark.go:32]: should annotation be changed instead to standard way of naming them, like: `gitpod.io/benchmark`?

File: dev/loadgen/cmd/benchmark.go [hunk 2]
```diff
@@ -57,8 +61,11 @@ var benchmarkCommand = &cobra.Command{
 			Id: "will-be-overriden",
 			Metadata: &api.WorkspaceMetadata{
 				MetaId:    "will-be-overriden",
-				Owner:     "00000000-0000-0000-0000-000000000000",
+				Owner:     "c0f5dbf1-8d50-4d2a-8cd9-fe563fa53c71",
```
Comment (sagor999) [inline dev/loadgen/cmd/benchmark.go:64]: is this just random uuid?

File: dev/loadgen/pkg/loadgen/executor.go [hunk 3]
```diff
@@ -161,10 +165,51 @@ func (w *WsmanExecutor) Observe() (<-chan WorkspaceUpdate, error) {
...
+			log.Warnf("failed to stop %s", id)
+		}
+	}
+
+	w.workspaces = make([]string, 0)
+
+	listReq := api.GetWorkspacesRequest{
+		MustMatch: &api.MetadataFilter{
+			Annotations: map[string]string{
+				"benchmark": "true",
```
Comment (sagor999) [inline dev/loadgen/pkg/loadgen/executor.go:191]: this is hard coded here instead of using constant

File: dev/loadgen/cmd/benchmark.go (hunk 1 above)
Comment (Furisto) [reply dev/loadgen/cmd/benchmark.go:32]: ws-manager will prefix it with gitpod.io/annotation. If you do it yourself, an error will occur.

File: dev/loadgen/cmd/benchmark.go (hunk 2 above)
Comment (Furisto) [reply dev/loadgen/cmd/benchmark.go:64]: yes

**Review by sagor999 — APPROVED** [review]

**Pull request merged**